pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -339,12 +340,16 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 	// Deploy work agent
-	// scale up the work agent deployment when the hubConnectionDegraded condition is False.
+	// scale up the work agent deployment when the hubConnectionDegraded condition is True and hub-kubeconfig-secret is missing.
```

Comment:
please doc more context why we need to check the reason?

Reply:
done